core/src/gl/shaderProgram.cpp

```diff
@@ -97,18 +98,21 @@ GLint ShaderProgram::getUniformLocation(const UniformLocation& _uniform) {
 }
 
 bool ShaderProgram::use() {
+    bool valid = true;
 
     checkValidity();
 
     if (m_needsBuild) {
-        build();
+        valid |= build();
```

Comment:
Actually it's not since the shader program id is tested on line `109`, will remove.